packages/tendermint-rpc/src/rpcclients/httpbatchclient.ts

```diff
@@ -68,27 +78,43 @@ export class HttpBatchClient implements RpcClient {
     }
   }
 
-  private async tick(): Promise<void> {
+  /**
+   * This is called in an interval where promise rejections cannot be handled.
+   * So this is not async and HTTP errors need to be handled by the queued promises.
+   */
+  private tick(): void {
     // Avoid race conditions
-    const queue = this.queue.splice(0, this.options.batchSizeLimit);
+    const batch = this.queue.splice(0, this.options.batchSizeLimit);
 
-    if (!queue.length) return;
+    if (!batch.length) return;
 
-    const request = queue.map((s) => s.request);
-    const raw = await http("POST", this.url, this.headers, request);
-    // Requests with a single entry return as an object
-    const arr = Array.isArray(raw) ? raw : [raw];
+    const requests = batch.map((s) => s.request);
+    const requestIds = requests.map((request) => request.id);
 
-    arr.forEach((el) => {
-      const req = queue.find((s) => s.request.id === el.id);
-      if (!req) return;
-      const { reject, resolve } = req;
-      const response = parseJsonRpcResponse(el);
-      if (isJsonRpcErrorResponse(response)) {
-        reject(new Error(JSON.stringify(response.error)));
-      } else {
-        resolve(response);
-      }
-    });
+    http("POST", this.url, this.headers, requests).then(
+      (raw) => {
+        // Requests with a single entry return as an object
+        const arr = Array.isArray(raw) ? raw : [raw];
+
+        arr.forEach((el) => {
+          const req = batch.find((s) => s.request.id === el.id);
+          if (!req) return;
+          const { reject, resolve } = req;
+          const response = parseJsonRpcResponse(el);
+          if (isJsonRpcErrorResponse(response)) {
+            reject(new Error(JSON.stringify(response.error)));
+          } else {
+            resolve(response);
+          }
+        });
+      },
+      (error) => {
+        for (const requestId of requestIds) {
+          const req = batch.find((s) => s.request.id === requestId);
+          if (!req) return;
+          req.reject(error);
+        }
+      },
```

Comment:
Those HTTP errors previosely got lost as unhandled promose rejections of the tick() call.